fee/src/lib.rs

```diff
@@ -28,9 +28,12 @@ pub fn calculate_fee_details(
     if zero_fees_for_test {
         return FeeDetails::default();
     }
-    let signature_fee = message
-        .num_total_signatures()
-        .saturating_mul(lamports_per_signature);
+    let signature_fee = (message
+        .num_transaction_signatures()
+        .saturating_add(message.num_ed25519_signatures())
+        .saturating_add(message.num_secp256k1_signatures())
+        .saturating_add(message.num_secp256r1_signatures()))
```

Comment:
wdyt adding a helper function to do all these `saturating_add`s, add comment to remind that this function need to be updated if signature types are removed/added in future, and feature gate is necessary?

Reply:
I'd kind of rather not, and would rather that all these signatures get removed from fees entirely w/ only the base `num_transaction_signatures` counting as "signatures" for fee calculations.

With `9ypxGLzkMxi89eDerRKXWDXe44UY2z4hBig4mDhNq5Dp` we're already moving towards precompiles being treated more similarly to other programs. I think it's natural we remove the special-case in fee to reflect that. Any "cost" to the validator should be reflected in additional CUs and require priority to make them worth processing.

Reply:
In shorter term, I am on the fence of removing `SVMMesage` from this interface entirely and having a `struct FeeSignatures {}` as an argument to this (defined in this crate).
Makes it much more clear exactly what is going into fee calculation.

Reply:
^ actually.. typing it out made it more clear to me. We should 100% make this change. It'd make it impossible for unexpected stuff from the much more generic `SVMMessage` interface to creep into the fee-calculation.

Reply:
yea, I agree. Was just thinking if putting these adds in one separate place would make it easier to change when precompiles become normal programs. Your suggestion of removing `SVMMessage` would serve same purpose.